src/main/java/nextstep/subway/domain/Line.java

```diff
@@ -129,86 +98,38 @@ public List<Station> getStations() {
         while (currSection != null) {
             stations.add(currSection.getUpStation());
             stations.add(currSection.getDownStation());
-            currSection = nextSection(currSection);
+            currSection = sections.getNextSection(currSection);
         }
 
         return new ArrayList<>(stations);
     }
 
-    private Section nextSection(Section currSection) {
-        // 다음 구간 : 현재 구간의 하행역이 상행역인 구간
-        return sections.stream().filter(sec -> sec.getUpStation().equals(currSection.getDownStation())).findFirst().orElse(null);
-    }
-
     public void removeSection(Station station) {
-        if (!isLastStation(station)) {
-            throw new IllegalArgumentException();
-        }
-
-        sections.remove(sections.size() - 1);
+        sections.removeSection(station, lastStation);
     }
 
     public int getLength() {
         if (!hasSections()) {
             return 0;
         }
 
-        return sections.stream().map(Section::getDistance).reduce(0, Integer::sum);
-    }
-
-    private boolean isFirstStation(Station station) {
-        return firstStation.equals(station);
-    }
-
-    private boolean isLastStation(Station station) {
-        return lastStation.equals(station);
+        return sections.getTotalDistance();
     }
 
-    private boolean isFirstSection(Section section) {
-        return isFirstStation(section.getDownStation());
+    public Station getFirstStation() {
+        return firstStation;
     }
```

Comment:
시작역과 종점역에 대한 정보는 구간정보에 포함되며 이는 Sections가 담당해야 할 영역으로 보이네요 

```suggestion
    public Station getFirstStation() {
        return sections.getFirstUpStation();
    }
```
하행역도 동일하게 이런식으로 책임 분리하는게 어떨까요?